README.md

```diff
@@ -160,21 +143,104 @@ try {
 }
 ```
 
+Some kinds of dependency fallbacks may be handled by [import maps](https://github.com/wicg/import-maps), but even with the support of import maps, [some scenarios](https://github.com/WICG/import-maps/blob/master/README.md#alternating-logic-based-on-the-presence-of-a-built-in-module) which are outside of the declaratively handled set would still benefit from top-level `await`.
+
+### WebAssembly Modules
+
+WebAssembly Modules are "compiled" and "instantiated" in a logically asynchronous way, based on their imports: Some WebAssembly implementations do nontrivial work at either phase, which is important to be able to shunt off into another thread. To integrate with the JavaScript module system, they will need to do the equivalent of a top-level await. See the [WebAssembly ESM integration proposal](https://github.com/webassembly/esm-integration) for more details.
+
+## Semantics as desugaring
+
+Top-level `await` makes importing a module automatically block on any top-level `await`s. You can think of it like, each module exports a Promise, and after all the `import` statements, but before the rest of the module, the Promises are all `await`ed:
+
+```mjs
+import { a } from './a.mjs'
+import { b } from './b.mjs'
+import { c } from './c.mjs'
+
+console.log(a, b, c);
+```
+
+would be equivalent to
+
+```mjs
+import aPromise, { a } from './a.mjs'
+import bPromise, { b } from './b.mjs'
+import cPromise, { c } from './c.mjs'
+
+Promise.all([aPromise, bPromise, cPromise]).then(() => {
```

Comment:
Is it worth including an `export const dPromise =` at the beginning of this line, to show how it propagates?